src/rust/engine/src/context.rs

```diff
@@ -171,10 +168,14 @@ impl Core {
       platform_properties: remote_execution_extra_platform_properties,
     };
 
+    // TODO: Propagate in a setting for this.
```

Comment:
Would the existing `local_store_dir` work? Have it be a parent for `lmdb_store` and `named_caches` with a default like `~/.cache/pants/local_execution`. I'm not seeing a good reason to have an independent toggle for this.